### PR TITLE
Revive sage live doc using jupyter-sphinx

### LINKED ISSUE #24367: Confusing documentation on MathJax in light of use in Jupyter notebook
<div id="comment:0"></div>

As pointed out to me in [this issue](https://github.com/sagemath/sage-windows/issues/9), [this documentation page](http://doc.sagemath.org/html/en/tutorial/latex.html) is a little confusing as to how use of the `MathJax` class relates to displaying LaTeX representations in the Jupyter notebook.

This confusion is increased by the fact that Jupyter doesn't display objects' latex repr by default (I'm not sure if that's the case or not for the classic Sage NB to which these docs were likely written).  The user was confused into believing that the repr of a `MathJax` object should actually be rendered, and not display the raw HTML.

This section should probably be rewritten since it's confusing usage instructions with implementation details that won't be interesting to a casual user.  I think these examples of using the `MathJax` class should be removed and/or moved to the class's API documentation (which could perhaps be linked to, but not further discussed?)

CC:  @slel @kwankyu

Component: **documentation**

Keywords: **sd111**

_Issue created by migration from https://trac.sagemath.org/ticket/24367_

### LINKED ISSUE #33320: Rework on live sage documentation
<div id="comment:0"></div>

As Thebe was not working in our sage documentation, we removed it from sage (see #33309, #33529) and introduced a new tool jupyter-sphinx in #33507.

As we move to the new tool, the ticket

- #24593 "Provide live documentation support in Jupyter using Thebe"

was closed.

Based on the efforts of #24593, the goal of this ticket is to deliver the live documentation using jupyter-sphinx so that every code blocks on every documentation page can be activated.

This ticket patches `jupyter-sphinx` with 
- https://github.com/kwankyu/jupyter-sphinx/compare/upstream...v0.3.2-sage-trac-33320?expand=1

Depends on #33763

CC:  @jhpalmieri

Component: **documentation**

Author: **Kwankyu Lee**

Branch/Commit: **[public/33320](https://github.com/sagemath/sagetrac-mirror/tree/public/33320) @ [`f5dff43`](https://github.com/sagemath/sagetrac-mirror/commit/f5dff4392a2551e9ae7634ecadccd53ec2158ac7)**

_Issue created by migration from https://trac.sagemath.org/ticket/33320_

### PR DESCRIPTION
**Pull request by kwankyu**
We fix #33320 and  #24367. 

We also upgrade jupyter-sphinx standard package to the latest version 0.4.0: https://github.com/jupyter/jupyter-sphinx/releases/tag/v0.4.0, and patch it to facilitate live doc.

By default, live doc depends on https://github.com/sagemath/sage-binder-env for binder (remote jupyter server). 

To showcase the live doc, we edited "A Tour of Sage"  doc and some parts of sage doc. Visit

- https://deploy-preview-36144--sagemath-tobias.netlify.app/a_tour_of_sage or 
- https://deploy-preview-36144--sagemath-tobias.netlify.app/tutorial/latex  (edited to resolve #24367)

**Click the activate button to activate live doc** The button is hidden to the right middle side of the browser. After clicked, the activate button shows the status of the launching kernel. 

**Be patient if the kernel loads slow.  It may take several minutes.** Sometimes it may be stuck in the "launching" state. In that case, reload the page and try again. 

Note that `%display latex` is the default. You should use `%display plain` to get plain textual output. 

## Preparation 

```
./sage -i jupyterlab
```

If things do not work as expected, first remove the existing documentation by
```
make doc-clean doc-uninstall
```

## Build static doc 

```
export SAGE_LIVE_DOC=no    # default
make -j4
```

## Build live doc with the default Binder repo 

``` 
export SAGE_LIVE_DOC=yes
export SAGE_JUPYTER_SERVER=binder    # default
make -j4
```

## Build live doc with a custom Binder repo

```
export SAGE_LIVE_DOC=yes
export SAGE_JUPYTER_SERVER=binder:sagemath/sage-binder-env    # default
make -j4
```
The kernel is "sagemath".

## Build live doc with a local server

```
export SAGE_LIVE_DOC=yes
export SAGE_JUPYTER_SERVER=http://localhost:8889
export SAGE_JUPYTER_SERVER_TOKEN=secret              # default 
make -j4
```
 and run a local jupyter server before browsing the documentation.
```
./sage --notebook=jupyterlab \
  --ServerApp.token='secret' \
  --ServerApp.allow_origin='null' \
  --ServerApp.disable_check_xsrf=true \
  --ServerApp.port=8889 \
  --ServerApp.open_browser=false 
```

## Build live doc selectively

First build static doc

```
make -j4
```
and then build live doc selectively either by

```
./sage --docbuild --live-doc a_tour_of_sage html
```
or
```
export SAGE_JUPYTER_SERVER=http://localhost:8889
export SAGE_JUPYTER_SERVER_TOKEN=secret              # default 
./sage --docbuild --live-doc a_tour_of_sage html
```
with a local server.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

Related upstream issues: 
- https://github.com/jupyter/jupyter-sphinx/issues/201
- https://github.com/jupyter/jupyter-sphinx/pull/231

### CODE REVIEW

**Comment by mkoeppe** [comment]
With furo in dark mode, it looks like this:
Screenshot 2023-08-28 at 10 47 45 AM

**Comment by kwankyu** [comment]
Try this.

**Comment by kwankyu** [comment]
This is ready for review.

**Comment by dimpase** [comment]
I've merged https://github.com/sagemath/sage-binder-env/pull/16 - can you use the official thing now?

**Comment by kwankyu** [comment]
Great! Thanks.

**Comment by dimpase** [comment]
shouldn't binder be mentioned in the docs in installation/#in-the-cloud ?

**Comment by dimpase** [comment]
how can this be tested?

**Comment by kwankyu** [comment]
> shouldn't binder be mentioned in the docs in installation/#in-the-cloud ?

Yes. I will write something in the installation guide.

But first we should decide which is default, live doc or static doc? In this branch, static doc is the default. But, in my opinion, there's no harm in setting live doc as default. Live doc becomes "live" only if you click on "Make live", and the time cost to build live doc is negligible, thanks to a small patch to jupyter-sphinx (and I also made a PR to submit the patch to the upstream https://github.com/jupyter/jupyter-sphinx/pull/231)

> how can this be tested? I tried "live" tour of Sage link, but all I get is "waiting for kernel" under a cell I tried

Binder is pulling the sage docker image to create the sage computing environment (container). It takes some time if Binder should do this from time to time. If live doc is the default and many people are using it, Binder will keep the container running and it would take no time to run the environment.

So keep it open, and do something else to pass some time before the cell runs :-) I just tried, it works well.

**Comment by kwankyu** [comment]
Making live doc the default would help it make Binder computing environment get run fast.

**Comment by dimpase** [comment]
ok, now it works for me too.

**Comment by dimpase** [comment]
I don't mind if you set the default to `live`.
Either way is fine with me.

**Comment by kwankyu** [comment]
> I don't mind if you set the default to `live`. Either way is fine with me.

OK. Then let's go with static doc as the default. Sage assumes a user without internet connection by default. That is the reason why `SAGE_USE_CDNS` is also "no" by default.

But we should set `export SAGE_LIVE_DOC=yes` every where in our development environment. in github, of course.

**Comment by kwankyu** [comment]
I made more edits than necessary in the installation guide while I am at it.

**Comment by kwankyu** [comment]
> how can this be tested? I tried "live" tour of Sage link, but all I get is "waiting for kernel" under a cell I tried

It seems we are affected by this

https://discourse.jupyter.org/t/something-up-with-mybinder-org-cache/18490

and this

https://blog.jupyter.org/mybinder-org-reducing-capacity-c93ccfc6413f

**Comment by kwankyu** [comment]
The workhorse of the live doc preview for PR is the binder repo

https://github.com/sagemath/sage-binder-env/tree/dev

This binder repo is updated to the latest develop version of Sage, daily. 

I made some efforts to reduce the time spent in"Waiting for kernel" before the doc goes live.

**Comment by kwankyu** [comment]
Loading time of the live doc mainly depends on the size of the docker image pulled by binder. It is now taking up to 4 minutes (the image is 12.73 GB on my system. Not sure if this is correct estimate -- it is too big) To reduce the time, it makes sense to write a Dockerfile creating smallest possible Sage docker image.

@mkoeppe, do you think there is room in reducing the Sage docker image?

**Comment by mkoeppe** [comment]
The Docker images at ghcr.io/sagemath/sage are not really meant for end users. 
They are huge because they have lots of layers from the build process, include the full logs, etc.

If you want a smaller image based on it, you can use `FROM ghcr.io/sagemath/sage/sage-ubuntu-focal-standard-with-system-packages` and then use `COPY --from` (https://docs.docker.com/engine/reference/builder/#copy) to copy the necessary parts from the `-with-targets` image. ("Multi-stage build")

**Comment by kwankyu** [comment]
Thanks for the tips!

**Comment by kwankyu** [comment]
> The Docker images at ghcr.io/sagemath/sage are not really meant for end users. They are huge because they have lots of layers from the build process, include the full logs, etc.
> 
> If you want a smaller image based on it, you can use `FROM ghcr.io/sagemath/sage/sage-ubuntu-focal-standard-with-system-packages` and then use `COPY --from` (https://docs.docker.com/engine/reference/builder/#copy) to copy the necessary parts from the `-with-targets` image. ("Multi-stage build")

Following the tips, I managed to reduce the sage docker image size from around 12.5 GB to 8 GB. The removed files are perhaps build artifacts and the doc. I feel that no further reduction could be possible.

Would you look through the Dockerfile to see if it is organized reasonably?

https://github.com/sagemath/sage-binder-env/blob/master/.github/Dockerfile

I wrote it by many trial-and-errors. So I am not sure if it is written "correct". In particular, I am not sure about lines from line 49.

8 GB is still huge for Binder. It looks struggling to launch the image.  But perhaps we should live with it, as sage is huge.

**Comment by mkoeppe** [comment]
This looks good. 

You could avoid some of the COPYing by using the installed script `/sage/venv/bin/sage` instead of `/sage/sage`.

**Comment by kwankyu** [comment]
> You could avoid some of the COPYing by using the installed script `/sage/venv/bin/sage` instead of `/sage/sage`.

I don't understand. There is no `/sage/venv/bin/sage` in the image "ghcr.io/sagemath/sage/sage-ubuntu-focal-standard-with-system-packages"... Which COPYing do you mean?

**Comment by kwankyu** [comment]
After 30% reduction of the docker image size, it still takes almost 5 minutes to pull the reduced image 
```
time docker pull ghcr.io/sagemath/sage-binder-env:10.2.beta1
```
while it takes 6 minutes for the vanilla image 
```
time docker pull ghcr.io/sagemath/sage/sage-ubuntu-focal-standard-with-targets:10.2.beta1
```
on my system. On Binder, it seemed to take around 4 minutes for the vanilla image (I don't know why).

Binder declares  an error if it fails to launch the jupyter environment in 300 seconds (= 5 minutes).

Hence this live doc idea relying on Binder seems a dead end. 

I wonder the (old) live doc ever worked well... @nthiery

**Comment by kwankyu** [comment]
One last remaining hope is that once we deploy live doc, many people use it and this accelerates launch time (and creates a virtuous circle :-)

**Comment by dimpase** [comment]
in principle we can self-host binder.

**Comment by kwankyu** [comment]
BinderHub? https://binderhub.readthedocs.io/en/latest/

That is beyond my horizon...

**Comment by dimpase** [comment]
> BinderHub? https://binderhub.readthedocs.io/en/latest/
> 
> That is beyond my horizon...

maybe @williamstein can advice on how hard this is to set up, and how much this might cost qua hosting.

**Comment by kwankyu** [comment]
If we have such a facility, then beyond live doc, we could use it for live testing PRs (a proof of concept is in https://github.com/sagemath/sage/pull/36245)

**Comment by mkoeppe** [comment]
For the gitpod deployment, for similar reasons we switched from Docker images based on the one on ghcr.io to installing Sage using conda-forge. See https://github.com/sagemath/sage/blob/develop/.gitpod.yml

**Comment by kwankyu** [comment]
Do you mean the approach adopted in

https://github.com/sagemath/sage-binder-env/tree/lab

? This is the `lab` branch of `sagemath/sage-binder-env`. From my experience, I am not sure if this works better.

What is the reason that this approach is expected to work better? I don't know.

Binder pulls the built (docker) image and runs jupyter, in either approach (conda and ghcr.io). As far as I understand, the difference is how the image is built. The size of the built image is at least the size of the OS (ubuntu) and the built Sage. The built Sage is 5.5 GB. The ubuntu with all system packages is around .5 GB.  Hence the built image must be of size at least 6 GB. This is huge for Binder, which expects images of size around 2 GB. (By the way, this calculation makes me wonder what accounts for 2 GB = 8 GB of the image built in ghcr.io approach - 6 GB)

**Comment by dimpase** [comment]
You can save about 1.6Gb by not including local/share/doc/ in the image

You can strip all the *.so and all the executable files. This saves a lot on each of them, e.g.
unstripped src/sage/libs/braiding.cpython-311-x86_64-linux-gnu.so is 1.2Mb, and after stripping it's only 112Kb.
This is going to shave another 1Gb (or perhaps even 1.5Gb)

**Comment by kwankyu** [comment]
> You can save about 1.6Gb by not including local/share/doc/ in the image

I already did it.

> You can strip all the *.so 

Really? Does Sage function without dynamically loading libraries?

> and all the executable files. 

Do you mean files in `bin/`? I want a sage fully functional to a user computing with sage.

On the other hand, I am reluctant to make a big list of removable files, which needs constant maintenance...

**Comment by dimpase** [comment]
> > You can save about 1.6Gb by not including local/share/doc/ in the image
> 
> I already did it.
> 
> > You can strip all the *.so
> 
> Really? Does Sage function without dynamically loading libraries?

this is removing _debugging info_ from binary libraries, cf [strip(1)](https://man7.org/linux/man-pages/man1/strip.1.html)
> 
> > and all the executable files.
> 
> Do you mean files in `bin/`? I want a sage fully functional to a user computing with sage.

Of course it will be fully functional, only "hard" errors such as segfaults will be less easy to debug. But the latter isn't what this is meant to be.

> 
> On the other hand, I am reluctant to make a big list of removable files, which needs constant maintenance...

just run `find . -name *.so.*` and `find sage/ -name *.so` to get such a list for dylibs (on Linux).
No maintainance needed.

We should create a `make` target `strip` to do this.

**Comment by mkoeppe** [comment]
That's `make micro_release`

**Comment by kwankyu** [comment]
Thanks for the tips. But after `make micro_release`, I get
```
$ ./sage
┌────────────────────────────────────────────────────────────────────┐
│ SageMath version 10.2.beta4, Release Date: 2023-09-24              │
│ Using Python 3.11.5. Type "help()" for help.                       │
└────────────────────────────────────────────────────────────────────┘
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Warning: this is a prerelease version, and it may be unstable.     ┃
┃ Warning: sage.all is not available; this is a limited REPL.        ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
Traceback (most recent call last):
  File "/Users/kwankyu/GitHub/sage-dev/src/bin/sage-ipython", line 12, in <module>
    from sage.repl.interpreter import SageTerminalApp
  File "/Users/kwankyu/GitHub/sage-dev/src/sage/repl/interpreter.py", line 144, in <module>
    from sage.repl.preparse import preparse, containing_block
  File "/Users/kwankyu/GitHub/sage-dev/src/sage/repl/preparse.py", line 246, in <module>
    from sage.repl.load import load_wrap
  File "/Users/kwankyu/GitHub/sage-dev/src/sage/repl/load.py", line 16, in <module>
    from sage.cpython.string import str_to_bytes, bytes_to_str, FS_ENCODING
ModuleNotFoundError: No module named 'sage.cpython.string'
```

**Comment by mkoeppe** [comment]
My guess would be that this has never been tested with editable builds

**Comment by kwankyu** [comment]
It seems so. Most of the code is from 2018.

**Comment by dimpase** [comment]
oops, not only a guard, a fix, too, I think

**Comment by dimpase** [comment]
> It seems so. Most of the code is from 2018.
> 
> The target `micro_release` needs a safe guard. It wiped out my `.git` directory!

How about splitting `strip` into a separate target (then you can do `make strip` without extra stuff removed)

```diff
--- a/Makefile
+++ b/Makefile
@@ -172,13 +172,16 @@ bootstrap-clean:
 maintainer-clean: distclean bootstrap-clean
        rm -rf upstream
 
+# stripping binaries - saves around 2Gb
+strip:
+       @echo "Stripping binaries ..."
+       LC_ALL=C find local/lib local/bin local/var/lib/sage/venv-python* -type f -exec strip '{}' ';' 2>&1 | grep -v "File format not recognized" |  grep -v "File truncated" || true
+
 # Remove everything that is not necessary to run Sage and pass all its
 # doctests.
-micro_release:
+micro_release: strip
        $(MAKE) sagelib-clean
        $(MAKE) misc-clean
-       @echo "Stripping binaries ..."
-       LC_ALL=C find local/lib local/bin local/var/lib/sage/venv-python* -type f -exec strip '{}' ';' 2>&1 | grep -v "File format not recognized" |  grep -v "File truncated" || true
        @echo "Removing sphinx artifacts..."
        rm -rf local/share/doc/sage/doctrees local/share/doc/sage/inventory
        @echo "Removing documentation. Inspection in IPython still works."
```

**Comment by kwankyu** [comment]
I am thinking of applying the stripping directly in the Dockerfile for Binder. But I could do this only slowly for lack of time.

Please open a new PR/issue if you want to revise `micro_release`. I may cherry-pick from it later :-)

**Comment by soehms** [comment]
> > You can strip all the *.so and all the executable files. This saves a lot on each of them, e.g. unstripped src/sage/libs/braiding.cpython-311-x86_64-linux-gnu.so is 1.2Mb, and after stripping it's only 112Kb. This is going to shave another 1Gb (or perhaps even 1.5Gb)
> 
> Yes! The size of the reduced image is 5.6 GB. Before it was 8.0 GB. `strip` is very powerful!
> 
> Just that 5.6 GB still seems large for Binder though...
> 
> Thanks!
> 
> This is the code used:
> 
> ```dockerfile
> # Slim Sage
> RUN LC_ALL=C find local src pkgs -executable -type f -exec strip '{}' ';' 2>&1 | grep -v "file format not recognized" || true
> 
> # To exclude doc directory in the next stage
> RUN rm -rf /sage/local/share/doc  
> ```

Currently there is a related problem with our Docker images on Docker Hub. Before the 10.2.beta7 release the new workflow [push_to_docker_hub.ym](https://github.com/sagemath/sage/blob/develop/.github/workflows/push_to_docker_hub.yml) produced [images of 9.9 GB](https://hub.docker.com/layers/sagemath/sagemath-dev/10.2.beta5/images/sha256-2ab4994430174b12e5986500682629ff6aaab34b1024f70c6254dcf63921011a?context=explore) for the `make-build` target and [of 3.1 GB](https://hub.docker.com/layers/sagemath/sagemath/develop/images/sha256-c227dda1d7f4ab9a44ed1ea9d520a6895ee7797e90986535183ae9bbd75230a4?context=explore) for the `make-micro-release` target (all uncompressed size).

Now, for 10.2.beta7 the `make-micro-release` failed with an *No space left on device* error (see also https://github.com/sagemath/sage/pull/36385#issuecomment-1769009000 and the [log-file](https://github.com/sagemath/sage/actions/runs/6520548141)). This is amazing since there [should be more than 25 GB available](https://github.com/marketplace/actions/maximize-build-disk-space-only-remove-unwanted-software) for the job. I could repeat the behavior when running the job for 10.2.beta7 in my own fork repository. What could be the reason for this? Maybe because the target `make-micro-release` is reached after an intermediate step `make-all`? From [docker/Dockerfile](https://github.com/sagemath/sage/blob/b7c1c8fe5cd0235a8c28d41d18a16dcbb9f10391/docker/Dockerfile#L218):


```
################################################################################
# Image with a full build of sage, ready to release.                           #
################################################################################
FROM make-all as make-release
RUN make micro_release
```

For what purpose do we need `make all` for these images?  Anyway, if I try to skip this using the target:

```
################################################################################
# Image with stripped build of sage, ready to release.                         #
################################################################################
FROM $MAKE_BUILD as make-micro
RUN make micro_release
```

with `docker build --build-arg MAKEFLAGS... --build-arg MAKE_BUILD="sagemath/sagemath-dev:develop" -f docker/Dockerfile --target=make-micro --tag sage .` I'm getting an even larger image with 10.4 GB.

Any ideas what I'm doing wrong here?

Furthermore: What could have increased the needed disk space since 10.2.beta5 (note that the image 10.2.beta6 failed because of [build errors](https://groups.google.com/g/sage-release/c/ow2oou9Q2dk/m/y6QJAJTuBQAJ))?

**Comment by kwankyu** [comment]
The "make live" button now shows the status of the launching kernel. Thus the user may wait for the kernel in an informed state. Try again in the built doc.

The Binder repo [sagemath/sage-binder-env](https://github.com/sagemath/sage-binder-env) providing the default kernel is automatically updated for the latest docker image. It is still slow but more or less stable.

Thus this PR seems complete as far as I am concerned. Please review. Let me know if there is anything to do (hopefully actionable).

**Comment by mkoeppe** [comment]
Just took a quick look and noticed that https://github.com/sagemath/sage/pull/36144#issuecomment-1696098790 is still unfixed

**Review by mkoeppe — COMMENTED** [review]
I have reviewed the changes to the installation guide, looking good.

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -102,6 +102,8 @@ jobs:
         run: |
           set -ex
           export SAGE_USE_CDNS=yes
+          export SAGE_LIVE_DOC=yes
+          export SAGE_JUPYTER_SERVER=binder:sagemath/sage-binder-env/dev
           mv /sage/local/share/doc/sage/html/en/.git /sage/.git-doc
```
Comment (mkoeppe) [inline .github/workflows/doc-build.yml]: I think we should activate it in the doc-build workflow only on pushes to release tags.

Elsewhere, in particular in a PR, I think it is problematic that the live doc uses a Sage version that is unrelated to the tested branch!

**Comment by kwankyu** [comment]
> Just took a quick look and noticed that [#36144 (comment)](https://github.com/sagemath/sage/pull/36144#issuecomment-1696098790) is still unfixed

Strange. Support for dark mode was fixed long ago, and It's been working fine for me. 

Screenshot 2023-10-25 at 7 28 30 AM

Would you check again? Some cookie problem...?

**Comment by mkoeppe** [comment]
To correct myself: Light mode and dark mode are fine, but the high-contrast mode (the third state of the toggle) is not.

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (kwankyu) [reply .github/workflows/doc-build.yml]: Right. 

I will change this.

**Comment by kwankyu** [comment]
There's no "high-contrast" mode for me... The third state is "automatic dark/light" (dual) mode for me.

**Comment by kwankyu** [comment]
@dima, the linter problem is fixed in https://github.com/sagemath/sage/pull/36522, which is a blocker and hence applied to this PR.

**Comment by mkoeppe** [comment]
> There's no "high-contrast" mode for me... The third state is "automatic dark/light" (dual) mode for me.

I think you may have explained this to me before. Anyway, I mean the third state, which has this pictogram shown here on the top right:
Screenshot 2023-10-24 at 4 20 35 PM

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (kwankyu) [reply .github/workflows/doc-build.yml]: How we will incorporate live doc to our ci (if ever) will be discussed separately. So I will remove the two lines.

But as this PR now needs review, I will keep the lines for now. If it is decided to merge this PR, then I will immediately push a commit to remove them.

**Comment by kwankyu** [comment]
I was referring to the same third state with icon "half white half dark". It is called "auto" mode. It seems that in auto mode, furo leaves the light/dark mode to the system. Perhaps our experiences in the auto mode differ because the system's settings are different.

The system's preference may be detected by `prefers-color-scheme` css media query.  I used that to fix for the auto mode in the commit ffa0af9. Let's see.

**Comment by kwankyu** [comment]
Now furo auto mode works for me. Please check on your side.

**Comment by kwankyu** [comment]
> I have reviewed the changes to the installation guide, looking good.

Thanks. I almost forgot what I did there. I will check again.

**Comment by mkoeppe** [comment]
> Now furo auto mode works for me. Please check on your side.

Unfortunately still the same.

I can reproduce it also using a screenshotting tool:
$ npx dark-mode-screenshot --url https://deploy-preview-36144--sagemath-tobias.netlify.app/tutorial/latex --output screenshot --fullPage --pause 750

**Comment by kwankyu** [comment]
The last commit patching jupyter-sphinx is to make `--live-doc` work as in `./sage --docbuild --live-doc tutorial html`.

**Comment by kwankyu** [comment]
> > Now furo auto mode works for me. Please check on your side.
> 
> Unfortunately still the same.

It's ready again. Would you check?

**Comment by mkoeppe** [comment]
Still the same

**Comment by kwankyu** [comment]
That is surprising!

I made it sure that in the furo auto mode, the change of the system light/dark mode setting takes effect. 
Screenshot 2023-10-27 at 1 53 03 AM

Screenshot 2023-10-27 at 1 52 50 AM

**Comment by mkoeppe** [comment]
which browser are you testing with? My tests are with Chrome on macOS; and the dark-mode-screenshot tool uses Chromium

**Comment by mkoeppe** [comment]
Same problem also with Safari on macOS

**Comment by kwankyu** [comment]
chrome on macos sonoma

**Comment by mkoeppe** [comment]
There's no problem on this page: https://deploy-preview-36144--sagemath-tobias.netlify.app/a_tour_of_sage/

The problem is with this page: https://deploy-preview-36144--sagemath-tobias.netlify.app/tutorial/latex

**Comment by kwankyu** [comment]
The same on safari:

Screenshot 2023-10-27 at 2 04 39 AM
Screenshot 2023-10-27 at 2 04 51 AM

**Comment by kwankyu** [comment]
It may be the cache problem... Would you turn this on "Disable cache"?

Screenshot 2023-10-27 at 2 10 28 AM

**Comment by mkoeppe** [comment]
The portion of the page that you screenshotted is also OK for me. The problem shows, as my previous screenshots have shown, a bit further down.

**Comment by kwankyu** [comment]
Yes, I got it! 

I have fixed this before... or no... I will check.

**Comment by kwankyu** [comment]
Basically I am fixing jupyter-sphinx that doesn't support dark mode. As the maintainer is not very responsive, we have to deal with that here.

Should be fixed now.

**Comment by kwankyu** [comment]
> The portion of the page that you screenshotted is also OK for me. The problem shows, as my previous screenshots have shown, a bit further down.

I missed that before...

**Comment by mkoeppe** [comment]
Yes! It's working!

**Review by mkoeppe — APPROVED** [review]
This is a great improvement. Let's merge it and make sure that it becomes available in our documentation for 10.2

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (mkoeppe) [reply .github/workflows/doc-build.yml]: To conditionalize its use in "build.yml" , one could use `$GITHUB_REF ` https://docs.github.com/en/actions/learn-github-actions/variables

**Comment by kwankyu** [comment]
As live doc is totally optional, it is left to users and the project how to use it. 

Currently a major obstacle is of course slow (and sometimes failing) kernel loading from the Binder repo. It seems that there is not much left to do in this regard on our side, and perhaps neither on Binder side too.

For individual users, running a local server is the best way to use live doc. Running and experimenting with the code examples right in the live doc would improve accessibility of the vast features of Sage. In this regard, many code examples should be refactored to facilitate live doc.

For the project, I am not sure. I hope that live doc be available in all online documentation, but because of the major obstacle, this needs more thought and discussion.

**Comment by kwankyu** [comment]
Thank you all!

**Pull request merged**